CONTRIBUTING.md

```diff
@@ -0,0 +1,169 @@
+
+- **You are not your code.** When your code is critiqued, questioned, or constructively criticized, remember that you are not your code. Do not take code review personally.
+
+- **Always do your best.** No one writes bugs on purpose. Do your best, and learn from your mistakes.
+
+- Kindly note any violations to the guidelines specified in this document.
+
+### 💅 Coding Style
+
+Consistency is the most important. Following the existing style, formatting, and naming conventions of the file you are modifying and of the overall project. Failure to do so will result in a prolonged review process that has to focus on updating the superficial aspects of your code, rather than improving its functionality and performance.
+
+For example, if all private properties are prefixed with an underscore `_`, then new ones you add should be prefixed in the same way. Or, if methods are named using camelcase, like `thisIsMyNewMethod`, then do not diverge from that by writing `this_is_my_new_method`. You get the idea. If in doubt, please ask or search the codebase for something similar.
+
+When possible, style and format will be enforced with a linter.
+
+## ⚙️ Setup project
+
+### Requirements
+
+- **PHP [8.1+](https://php.net/releases/)**
+
+### Installation
+
+1. Clone project
+2. Install dependencies
+
+```bash
+composer install
+```
+
+<details>
+<summary>
+<h3>Commands</h3>
+</summary>
+
+Keep a modern codebase with [**symplify/easy-coding-standard**](https://github.com/easy-coding-standard/easy-coding-standard)
+
+```bash
+composer style:fix
+```
+
+⚗️ Run static analysis using [**PHPStan**](https://phpstan.org/user-guide/command-line-usage)
+
+```bash
+composer phpstan
+```
+
+✅ Run unit tests using [**PEST**](https://pestphp.com/docs/writing-tests)
+
+```bash
+composer pest
+```
+
+💽 Analyzes your codebase and generates a report of your [type coverage](https://pestphp.com/docs/type-coverage) using [**PEST**](https://pestphp.com/docs/writing-tests)
+
+```bash
+composer pest:type-coverage
+```
+
+📈 Expose [code coverage](https://pestphp.com/docs/test-coverage) using [**PEST**](https://pestphp.com/docs/test-coverage) (_require XDEBUG in `coverage` mode_)
+
+```bash
+composer pest:coverage
+```
+
+🚀 Run the entire test suite
+
+```bash
+composer test
+```
+
+🧰 Update dependencies and validate `composer.json`
+
+```bash
+composer core:update
+```
+
+✅ Run refactors using [**Rector**](https://getrector.com/documentation)
+
+```bash
+composer rector
+```
+
+</details>
+
+## 🏅 Contributor Agreement
+
+By making a contribution to this project, I certify that I agree to the [Developer's Certificate of Origin 1.1](https://developercertificate.org/):
+
+To confirm that you have read this guide and are following it as best as possible, **include this emoji at the top** of your issue or pull request: 🍱
```

Comment:
I feel like this won't ever be enforced, it can be removed.
```suggestion
```

👉 I think it's best to add templates with a checkbox stating "[] I confirm that I have read the [Contributing Guide](./../CONTRIBUTING.md) and [Security Guide](./../SECURITY.md)" (as you have done with the pull request template).